### DISCUSSION TITLE
Hive Connector - INTERNAL ERROR — NO_NODES_AVAILABLE

### DISCUSSION THREAD

**Discussion by vineetalagh** [trinodb/trino#23248] [Q&A] [answered]
HI,

 I have got hive connector setup for s3 csv files. After upgrading to Version 454, each query to external csv table throws NO_NODES_AVAILABLE error.

`2024-09-02T22:55:29.654Z        INFO    dispatcher-query-6      io.trino.event.QueryMonitor     TIMELINE: Query 20240902_225527_00013_ybu3e :: FAILED (NO_NODES_AVAILABLE) :: elapsed 2288ms:: planning 185ms :: waiting 88ms :: scheduling 2103ms :: running 0ms :: finishing 2103ms :: begin 2024-09-02T22:55:27.228Z :: end 2024-09-02T22:55:29.516Z`

Other catalog connection like database and iceberg are running fine. Worker pods have enough memory and no resource contention. Even running Hive query on an empty internal s3 table  returns the same error.

Any idea what could be the issue?

**Comment by shauryagoel** [comment]
Facing the same issue on trino 454 when adding new catalogs and redeploying using helm.

**Comment by hashhar** [comment, marked as answer]
are the catalog configs present on all nodes?

Check the worker logs to see if the catalog was loaded on the workers or not.

**Reply by shauryagoel** [reply]
The worker pod has the catalog properties file. But, the worker has not loaded the new catalog. 
When I redeployed the helm chart only the coordinator restarted. Also in the worker logs I start getting-
```
2024-09-03T12:08:11.978Z        INFO    node-state-poller-0     io.trino.metadata.DiscoveryNodeManager  Previously active node is missing: trino-coordinator-57965478c5-97w99 (last seen at <IP>)
2024-09-03T12:08:11.978Z        INFO    node-state-poller-0     io.trino.metadata.DiscoveryNodeManager  Previously active node is missing: trino-worker-65c4798f54-m49zz (last seen at <IP>)
```
Previous catalogs work fine though.
